### PR TITLE
Rename `COWMap` to `CopyOnWriteMap` and add javadoc

### PR DESCRIPTION
**Pull request by snuyanzin**

### CODE REVIEW

File: src/main/java/net/datafaker/internal/helper/CopyOnWriteMap.java [hunk 1]
```diff
@@ -5,11 +5,24 @@
 import java.util.Set;
 import java.util.function.Supplier;
 
-public class COWMap<K, V> implements Map<K, V> {
+/**
+ * This is a Copy On Write map. The main idea behind is that
+ * there is lots of static info per provider to make this provider working.
+ * At the same time there is no need to load all this info at start since
+ * we don't know which roviders will be used and loading for all takes time.
```
Comment (asolntsev) [inline src/main/java/net/datafaker/internal/helper/CopyOnWriteMap.java:12]: typo: "roviders"

File: src/main/java/net/datafaker/internal/helper/CopyOnWriteMap.java [hunk 2]
```diff
@@ -5,11 +5,24 @@
...
+ * For that reason it is loaded on request and stores in these Copy On Write maps.
+ * Since it is loaded only once per provider and then is only read then
```
Comment (asolntsev) [inline src/main/java/net/datafaker/internal/helper/CopyOnWriteMap.java:14]: too many "then"s :)

**Review by asolntsev — CHANGES_REQUESTED** [review]
few typos

File: src/main/java/net/datafaker/internal/helper/CopyOnWriteMap.java (hunk 1 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/internal/helper/CopyOnWriteMap.java]: fixed

File: src/main/java/net/datafaker/internal/helper/CopyOnWriteMap.java (hunk 2 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/internal/helper/CopyOnWriteMap.java]: let's replace one of them)

**Review by asolntsev — APPROVED** [review]

**Comment by kingthorin** [comment]
I have a few more tweaks coming for this one. Just need to get in front of a computer. It'll be a few more hours.

**Comment by kingthorin** [comment]
Sorry I didn't manage it this evening because a school thing went long. I'll tackle it in the morning.

It's just minor tweaks to the Java doc. I think I can do it as a suggestion and just apply it then approve and merge.

**Comment by snuyanzin** [comment]
Ok, will wait for your response @kingthorin 
currently there is no rush
thanks for looking and willingness to improve it

**Comment by bodiam** [comment]
@kingthorin @snuyanzin I'll just merge it for now, if it's doc improvements, feel free to send a new PR!

**Comment by kingthorin** [comment]
Okay no worries. I'll send a new one.

**Pull request merged**